src/main/java/com/icthh/xm/ms/entity/web/rest/XmEntitySpecResource.java

```diff
@@ -58,19 +81,9 @@
     @PostFilter("hasPermission({'returnObject': filterObject, 'log': false}, 'XMENTITY_SPEC.GET')")
     public List<TypeSpec> getTypeSpecs(@ApiParam XmEntitySpecResource.Filter filter) {
         log.debug("REST request to get a list of TypeSpec");
-        List<TypeSpec> typeSpecs;
-        switch (filter != null ? filter : Filter.ALL) {
-            case APP:
-                typeSpecs = xmEntitySpecService.findAllAppTypes();
-                break;
-            case NON_ABSTRACT:
-                typeSpecs = xmEntitySpecService.findAllNonAbstractTypes();
-                break;
-            default:
-                typeSpecs = xmEntitySpecService.findAllTypes();
-                break;
-        }
-        return typeSpecs;
```

Comment:
look's nice.
It's really great!

Reply:
thnx

Reply:
@sergeysenja1992 take a look after @amedvedchuk core refactoring. It become more simple and readable.